terraform/modules/compute_plane/outputs.tf

```
@@ -0,0 +1,3 @@
+output "polling_agent_name" {
+  value = local.polling_agent_name
```

This would make Terraform aware of the dependency between the resource and the output.
You could even remove your local value with that.

```suggestion
  value = docker_container.polling_agent.name
```